components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -504,7 +508,10 @@ protected void processAuthenticationResponse(HttpServletRequest request,
             }
         }
         String password = request.getParameter(BasicAuthenticatorConstants.PASSWORD);
-
+        if (password.isEmpty()) {
```

Comment:
```suggestion
        if (StringUtils.isBlank(password)) {
```

Reply:
isBlank check for null and whitespaces as well. But  someone can have that as the password.

Reply:
`password.isEmpty()` could throw an NPE if password is null.